src/main/Main.hs

```diff
@@ -773,7 +773,7 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
                 -- Add RTS options to arguments
                 let argsWithRts args = if null eoRtsOptions 
                             then args :: [String]
-                            else args ++ ["+RTS"] ++ eoRtsOptions ++ ["-RTS"]
+                            else args ++ ["+RTS"] ++ concatMap words eoRtsOptions ++ ["-RTS"]
```

Comment:
It makes sense for --package, since (until the script command) it only took one package at a time, and even in the script command case, it doesn't follow normal escaping rules.

The --ghc-options though concerns me. Perhaps the String value is being parsed afterwards. There are a few different --ghc-options at play. Which one were you looking at?